### PR TITLE
[10.0][ADD] website_sale_price_subtotal: Show subtotal on product page instead of unit price

### PR DESCRIPTION
**Pull request by kellylougheed**
- Changes the price on the product page to the reflect the subtotal based on quantity rather than the unit price.

### CODE REVIEW

File: .travis.yml
Comment (lasley) [inline .travis.yml]: These changes can be reverted - it is [now in MQT](https://github.com/OCA/maintainer-quality-tools/blob/5b4b67e757611ac393aaf4434195ca5aa0cd4502/travis/travis_install_nightly#L35)

File: website_sale_price_subtotal/README.rst
Comment (lasley) [inline website_sale_price_subtotal/README.rst]: Remove comment

File: website_sale_price_subtotal/models/__init__.py
Comment (lasley) [inline website_sale_price_subtotal/models/__init__.py]: Update copyright to LasLabs Inc - SMD turned it over

File: website_sale_price_subtotal/__manifest__.py
Comment (lasley) [inline website_sale_price_subtotal/__manifest__.py]: Update author to LasLabs

File: website_sale_price_subtotal/__manifest__.py
Comment (lasley) [inline website_sale_price_subtotal/__manifest__.py]: superfluous - web is a dependency of website

File: website_sale_price_subtotal/__manifest__.py
Comment (lasley) [inline website_sale_price_subtotal/__manifest__.py]: trailing comma

File: website_sale_price_subtotal/models/product_product.py
Comment (lasley) [inline website_sale_price_subtotal/models/product_product.py]: Please double check this logic - it seems somewhat roundabout

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by lasley** [comment]
@obulkin please take this over

**Comment by obulkin** [comment]
I've updated this PR as follows:
* Redesign core logic to use JS instead of model changes so as to eliminate unintended side effects and preserve backend interfaces
* Add demo data and tour for testing new logic
* Update README
* Make style/organizational changes to bring module closer to OCA guidelines

It should now be ready for review.

**Comment by obulkin** [comment]
The latest commit expands the functionality in this module to cover the cart and the website checkout payment page. It should be ready for review again.

**Comment by lasley** [comment]
Hey @obulkin - We caught an issue with the rounding on the functional review. This is due to us using a decimal precision on pricing of 9 in order to make the rounding get to even numbers on the subtotals of the tiers. Example below, you can also inspect the SMD prod instance for more info
image

image

**Comment by lasley** [comment]
Also, looks like we have the subtotal crossed out & the unit price displayed in the add to cart dialog: 

image

**Comment by obulkin** [comment]
@lasley - I've just pushed up a commit that should address any rounding and modal cart bugs in this PR. It's now ready for review again.

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
Thanks @obulkin

**Comment by rafaelbn** [comment]
Hi @kellylougheed could you please review this? I will review and merge is all boots are green :smile: 


```
  File "/.repo_requirements/odoo/odoo/models.py", line 1083, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
ParseError: "Error while validating constraint

Element '<xpath expr="//span[@t-field='line.price_unit']">' cannot be located in parent view

Error context:
View `Website Sale - Payment Page Subtotals`
[view_id: 765, xml_id: n/a, model: n/a, parent_id: 720]
None" while parsing None:21, near
<data name="Website Sale - Payment Page Subtotals" inherit_id="website_sale.payment">
        <xpath expr="//thead//th[last()]" position="replace">
            <th width="130" class="text-center">Product Subtotal</th>
        </xpath>
        <xpath expr="//span[@t-field='line.price_unit']" position="attributes">
            <attribute name="t-field">line.price_subtotal</attribute>
        </xpath>
    </data>
```

**Comment by rafaelbn** [comment]
Dear @fgi-odoo this is a "Must have" in Odoo core. Any ecommerce have this

cc @tecnativa

**Comment by fgi-odoo** [comment]
Debatable. When we redesigned the checkout, we deeply thought about that. 
We wanted the cleanest screen possible. We figured out that the unit price was the most important piece of product information, compared to product total. As you provide the order total underneath.
We also compared with other major ecommerce software. Some like Shopify (ttp://i.imgur.com/DFtcPFZ.png) are doing like us. 

There is very little feedback about it. So I don't think we will change that anytime soon.

**Comment by rafaelbn** [comment]
Dear @kellylougheed ,

Are you going to finish this PR? :smile: I can review it after all bots are green in order to merge

If not we should close it :cry: 

Thanks

**Comment by pedrobaeza** [comment]
Closing as this was part of LasLabs, so no expected retake.

**Pull request closed without merging**
